Mathlib/LinearAlgebra/RootSystem/Defs.lean

```diff
@@ -547,11 +575,17 @@ def coxeterWeight : R := pairing P i j * pairing P j i
 lemma coxeterWeight_swap : coxeterWeight P i j = coxeterWeight P j i := by
   simp only [coxeterWeight, mul_comm]
 
-lemma exists_int_eq_coxeterWeight [P.IsCrystallographic] (i j : ι) :
-    ∃ z : ℤ, P.coxeterWeight i j = z := by
-  obtain ⟨a, ha⟩ := P.exists_int i j
-  obtain ⟨b, hb⟩ := P.exists_int j i
-  exact ⟨a * b, by simp [coxeterWeight, ha, hb]⟩
+/-- A variant of `RootPairing.coxeterWeight` for root pairings which are valued in a smaller set of
+coefficients.
+
+Note that it is uniquely-defined only when the map `S → R` is injective, i.e., when we have
+`[NoZeroSMulDivisors S R]`. -/
+def coxeterWeightIn (S : Type*) [CommRing S] [Algebra S R] [P.IsValuedIn S] (i j : ι) : S :=
+  P.pairingIn S i j * P.pairingIn S j i
+
+@[simp] lemma algebraMap_coxeterWeightIn (S : Type*) [CommRing S] [Algebra S R] [P.IsValuedIn S] :
```

Comment:
It doesn't matter too much, but I think it is nice if `i j` come after `S` in this lemma as well.